src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -312,7 +314,12 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
          _: PLess | _: PAtMost | _: PGreater | _: PAtLeast =>
       BooleanT
 
-    case _: PAdd | _: PSub | _: PMul | _: PMod | _: PDiv | _: PLength => IntT
+    case _: PIntLit | _: PAdd | _: PSub | _: PMul | _: PMod | _: PDiv =>
+      val exprNum = expr.asInstanceOf[PNumExpression]
```

Comment:
Can it not infer that on its own, i.e. did you try to remove the as instance of?

Reply:
No, it leads to a Type Mismatch because expr is of type `PActualExpression`.

Reply:
Replaced this part of code by
```scala
case exprNum: PNumExpression =>
      val typ = intExprType(exprNum)
      if (typ == IntT(UntypedConst)) getDefaultType(exprNum).getOrElse(typ) else typ
```